_extensions/bluesky-comments/bluesky-comments.js

```diff
@@ -118,19 +124,37 @@ class BlueskyComments extends HTMLElement {
     }
   }
 
-  #convertUri(uri) {
+  #convertToAtProtoUri(uri) {
     if (uri.startsWith('at://')) return uri;
 
     const match = uri.match(/profile\/([\w.]+)\/post\/([\w]+)/);
     if (match) {
-      const [, did, postId] = match;
-      return `at://${did}/app.bsky.feed.post/${postId}`;
+      const [, did, rkey] = match;
+      return this.createAtProtoUri({ did, rkey });
     }
 
     this.error = 'Invalid Bluesky post URL format';
     return null;
   }
 
+  createAtProtoUri({ did, rkey }) {
+    return `at://${did}/app.bsky.feed.post/${rkey}`;
+  }
+
+  #convertToHttpUrl(uri) {
+    uri = this.#convertToAtProtoUri(uri);
+    const [, , profile, , rkey] = uri.split('/');
+    return this.createPostUrl({ profile, rkey });
+  }
+
+  createPostUrl({ profile, rkey }) {
+    profile = profile || this.profile;
+    if (profile.startsWith('@')) {
+      profile = profile.slice(1);
+    }
+    return `https://bsky.app/profile/${profile}/post/${rkey}`;
```

Comment:
Currently hard-wired to `https://bsky.app` as the base URL, but we'll probably need to think more carefully about this in the future